`Dependency` New Resource azurerm_healthcare_fhir_service (#15913)
hashicorp/terraform-provider-azurerm: website/docs/r/healthcare_fhir_service.html.markdown

* `name` - (Required) Specifies the name of the Healthcare FHIR Service. Changing this forces a new Healthcare FHIR Service to be created.

* `workspace_id`  - (Required) Specifies the name of the Healthcare Workspace where the Healthcare FHIR Service should exist. Changing this forces a new Healthcare FHIR Service to be created.

* `location` - (Required) Specifies the Azure Region where the Healthcare FHIR Service should be created. Changing this forces a new Healthcare FHIR Service to be created.

* `kind` - (Required) Specifies the kind of the Healthcare FHIR Service. Possible values are: `fhir-Stu3` and `fhir-R4`. Defaults to `fhir-R4`. Changing this forces a new Healthcare FHIR Service to be created.

* `identity` - (Optional) An `identity` block as defined below.

* `access_policy_object_ids` - (Optional) A list of the access policies of the service instance.

* `cors_configuration` - (Optional) A `cors_configuration` block as defined below.

* `acr_login_servers` - - (Optional) A list of azure container registry settings used for convert data operation of the service instance.

* `authentication` - (Required) An `authentication` block as defined below.

> * `export_storage_account_name` - (Optional) specifies the name of the export storage account which accepts the operation configuration information

Reviewer: grammer, caps, punciation
```suggestion
* `export_storage_account_name` - (Optional) Specifies the name of the storage account which the operation configuration information is exported to.
```

please fix all documentation grammer, caps and punctuation.

also this property is unclear, maybe it should be `configuration_export_storage_account_name`

* `public_network_access_enabled` - (Optional) Whether to enabled public networks when data plane traffic coming from public networks while private endpoint is enabled.

---
An `identity` block supports the following:

* `type` - (Required) The type of identity used for the Healthcare FHIR service. Possible values are `SystemAssigned`.

---
A `cors_configuration` block supports the following:  

* `allowed_origins` - (Required) A set of origins to be allowed via CORS.
* `allowed_headers` - (Required) A set of headers to be allowed via CORS.
* `allowed_methods` - (Required) The methods to be allowed via CORS.
* `max_age_in_seconds` - (Required) The max age to be allowed via CORS.

> * `allow_credentials` - (Boolean) If credentials are allowed via CORS.

Reviewer: could we make this
```suggestion
* `credentials_allowed` - (Boolean) If credentials are allowed via CORS.
```

Reviewer: also we do not add boolean here, this should be optional or required
```suggestion
* `allow_credentials` - (optional|required) If credentials are allowed via CORS.
```

Author: I used the same property definition from the existing resource [healthcare_service](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/healthcare_service), should I change that resource too?

Reviewer: yes please
